aten/src/ATen/cpu/vec/vec_base.h

```diff
@@ -839,6 +847,12 @@ inline Vectorized<T>& operator <<= (Vectorized<T>& a, const Vectorized<T>& b) {
   return a;
 }
 
+template <typename T>
+inline Vectorized<T>& operator >>= (Vectorized<T>& a, const Vectorized<T>& b) {
+  a = a >> b;
+  return a;
+}
+
```

Comment:
I see that all the functions use this pattern. It's weird to me that we don't use the usual C++ pattern of implementing the out-of-place op in terms of the in-place one.
@peterbell10 any idea why do we do this?

Reply:
Probably because the intrinsic operations are naturally out-of-place.